modules/nlu/src/backend/pipelines/entities/patternExtractor.ts

```diff
@@ -47,3 +47,33 @@ export const extractPatternEntities = (input: string, entityDefs: sdk.NLU.Entity
     }))
   })
 }
+
+const _extractEntitiesFromOccurence = (candidate: string, occurence: sdk.NLU.EntityDefOccurence): sdk.NLU.Entity[] => {
+  const pattern = RegExp([occurence.name, ...occurence.synonyms].join('|'), 'i')
```

Comment:
you need to escape the patterns here, see https://github.com/botpress/botpress/blob/70b5020595c5df6c7afccbfde53c6077a7f679d9/modules/nlu/src/backend/pipelines/intents/matcher.ts#L1